timewarp_gl/plugin.cpp

```diff
@@ -64,7 +65,27 @@ class timewarp_gl : public threadloop {
 		  // In production systems, this is certainly a good thing, but it makes the system harder to analyze.
 		, disable_warp{ILLIXR::str_to_bool(ILLIXR::getenv_or("ILLIXR_TIMEWARP_DISABLE", "False"))}
 		, enable_offload{ILLIXR::str_to_bool(ILLIXR::getenv_or("ILLIXR_OFFLOAD_ENABLE", "False"))}
-	{ }
+	{ 
+		image_handles_ready = false;
+		swapchain_ready = false;
+		sb->schedule<image_handle>(id, "image_handle", [this](switchboard::ptr<const image_handle> handle, std::size_t) {
+			if (handle->swapchain_index > 1) {
+				return;
+			}
+			
+			assert(handle->swapchain_index == 0 || handle->swapchain_index == 1);
+
+
+			// If the graphics backend is OpenGL, Monado's OpenGL context should be
+			// shared with ILLIXR, so the buffers are the same. 
+			this->_m_image_handles[handle->swapchain_index].push_back(*handle);
+
+			if (this->_m_image_handles[0].size() == (size_t) handle->num_images &&
+				this->_m_image_handles[1].size() == (size_t) handle->num_images) {
+				this->image_handles_ready = true;
```

Comment:
There is a data race here. `image_handles_ready` is being written here and being read on line 428. Since the two are happening in two different threads, and the two threads can be running in parallel, and there is no other synchronization between the two threads, `image_handles_ready` needs to be an atomic bool.